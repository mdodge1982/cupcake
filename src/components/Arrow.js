import React, {Component} from 'react';
import './Arrow.css';

class Arrow extends Component {
	constructor() {
		super();
		this.state = {
			int: null
		};
	}
	render() {
		let className = 'Arrow ' + this.props.direction;
		return (
			<button
				className={className}
				onMouseDown={(e) => this.onMouseDown(e)}
				onMouseUp={(e) => this.onMouseUp(e)}
				onTouchStart={(e) => this.onMouseDown(e)}
				onTouchEnd={(e) => this.onMouseUp(e)}>
			</button>
		);
	}
	onMouseDown() {
		const int = setInterval(() => {
			this.props.moveBag(this.props.direction);
		},50);
		this.setState({int:int});
	}
	onMouseUp() {
		clearInterval(this.state.int);
		this.props.stopBag();
	}
}

export default Arrow;
