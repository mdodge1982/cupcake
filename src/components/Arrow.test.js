import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import renderer from 'react-test-renderer';
import Arrow from './Arrow';

const reducer = (state = {}) => state;
const store = createStore(reducer);

test('Render right arrow', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Arrow direction="right" />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Render left arrow', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Arrow direction="left" />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

jest.useFakeTimers();

test('Mouse event test', () => {
	const mockMove = jest.fn();
	const mockStop = jest.fn();
	const component = ReactTestUtils.renderIntoDocument(
		<Arrow direction="left" store={store} moveBag={mockMove} stopBag={mockStop} />
	);
	const button = ReactTestUtils.findRenderedDOMComponentWithTag(component,'button');
	ReactTestUtils.Simulate.mouseDown(button);
	jest.runTimersToTime(51);
	ReactTestUtils.Simulate.mouseUp(button);
	expect(mockMove.mock.calls.length).toBe(1);
	expect(mockMove.mock.calls[0][0]).toBe('left');
	expect(mockStop.mock.calls.length).toBe(1);
});
