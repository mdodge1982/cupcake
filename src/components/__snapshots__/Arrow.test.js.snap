// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render left arrow 1`] = `
<button
  className="Arrow left"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
/>
`;

exports[`Render right arrow 1`] = `
<button
  className="Arrow right"
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
/>
`;
